translate/filters/checks.py

```diff
@@ -309,6 +309,10 @@ def __init__(self, checkerconfig=None, excludefilters=None,
         else:
             self.setconfig(checkerconfig)
 
+        # Exclude extra tests from being treated as tests. This is used to
+        # include some of these extra tests based on language config.
```

Comment:
Not 100% sure I understand what this part is trying to achieve.  If they're included based on language why not call it `lang_specific_tests` or similar? Or `extra_optional_checks` Or `extra_conditional_checks`